### PR TITLE
grt: coverity and bazel build fixes

### PR DESCRIPTION
**Pull request by eder-matheus**
## Summary
- `grt`: Bazel `fastroute` target was missing `-fopenmp` and `@openmp` dep, so OpenMP pragmas were silently dropped — added them.
- `grt`: Fixed Coverity CID 1646072 in `RipUp.cpp::removeEdgeFromNode` — `BIG_INT` (1e9) was being assigned to `int16_t botL`/`topL` when a node's last edge was removed, causing truncation. Now resets to constructor defaults (-1) when `conCNT` reaches 0.

## Type of Change
- Bug fix

## Impact
- Bazel-built `fastroute` now parallelizes correctly (snapshot-batched routing waves and the existing OpenMP loops in `FastRoute.cpp`); previously ran single-threaded regardless of `num_threads_`. CMake builds unchanged.
- `removeEdgeFromNode`: when a TreeNode's last edge is removed, layer/edge fields are now `-1` instead of truncated `BIG_INT` garbage. Edge case during ripup of a Steiner node with a single connection.

## Verification
- [x] I have verified that the local build succeeds (`./etc/Build.sh`).
- [x] I have run the relevant tests and they pass.
- [x] My code follows the repository's formatting guidelines.
- [x] **I have signed my commits (DCO).**

## Related Issues
Coverity CID 1646072

### CODE REVIEW

File: src/grt/src/fastroute/src/RipUp.cpp [hunk 1]
```diff
@@ -302,10 +302,18 @@ bool FastRouteCore::newRipup3DType3(const int netID, const int edgeID)
...
-    node.botL = bl;
-    node.lID = bid;
-    node.topL = hl;
-    node.hID = hid;
+    if (node.conCNT > 0) {
+      node.botL = bl;
+      node.lID = bid;
+      node.topL = hl;
+      node.hID = hid;
+    } else {
```
[the remark this answers is not in the document]
Comment (eder-matheus) [reply src/grt/src/fastroute/src/RipUp.cpp]: The suggestion is based on an incorrect type assumption. `lID` and `hID` are declared as `int`, not `int16_t`. Only `botL` and `topL` are 16-bit, which is exactly the case Coverity flagged (CID 1646072) and that this PR addresses. Assigning `BIG_INT` (1e9) to a 32-bit int is well-defined and does not truncate.

**Comment by maliberty** [comment]
failed tests

**Comment by eder-matheus** [comment]
> failed tests

Should be fixed now.

**Review by maliberty — APPROVED** [review]
